File: Components/Reporting/CS/ASPxReportDemos/UserControls/Search.ascx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.HtmlControls;
using System.Web.UI.WebControls;
using DevExpress.Web;
using DevExpress.Web.Demos;

public partial class UserControls_Search : System.Web.UI.UserControl {
    public bool FocusOnLoad { get; set; }

    protected void Page_Load(object sender, EventArgs e) {
        if(FocusOnLoad)
            TbSearch.Focus();

        if (Utils.IsAccessibilityDemo)
            TbSearch.NullText = "Search";
    }
    protected void SearchPopup_WindowCallback(object source, DevExpress.Web.PopupWindowCallbackArgs e) {
        var text = e.Parameter;
        var results = SearchUtils.DoSearch(text);
        var resultsPanel = SearchPopup.FindControl("ResultsPanel");
        var noResultsPanel = SearchPopup.FindControl("NoResultsPanel");
        if(results.Count > 0) {
            resultsPanel.Visible = true;
            noResultsPanel.Visible = false;
            var grouppedResults = results.GroupBy(sr => sr.Product);
            foreach(var productGroup in grouppedResults) {
                var product = productGroup.Key;
                var group = SearchResultsNavBar.Groups.Add(product.NavItemTitle.ToUpper(), string.Empty, string.Empty, Utils.GenerateDemoUrl(product.Intro));
                foreach(var res in productGroup) {
                    group.Items.Add(res.Text, string.Empty, string.Empty, Utils.GenerateDemoUrl(res.Demo));
                }
            }
        }
        else {
            resultsPanel.Visible = false;
            noResultsPanel.Visible = true;
            var requestText = noResultsPanel.FindControl("RequestText") as HtmlGenericControl;
            requestText.InnerHtml = text;
        }
    }
}

File: Components/Reporting/VB/ASPxReportDemos/UserControls/Search.ascx.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.HtmlControls
Imports System.Web.UI.WebControls
Imports DevExpress.Web
Imports DevExpress.Web.Demos

Partial Public Class UserControls_Search
	Inherits System.Web.UI.UserControl
	Public Property FocusOnLoad() As Boolean

	Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs)
		If FocusOnLoad Then
			TbSearch.Focus()
		End If

		If Utils.IsAccessibilityDemo Then
			TbSearch.NullText = "Search"
		End If
	End Sub
	Protected Sub SearchPopup_WindowCallback(ByVal source As Object, ByVal e As DevExpress.Web.PopupWindowCallbackArgs)
		Dim text = e.Parameter
		Dim results = SearchUtils.DoSearch(text)
		Dim resultsPanel = SearchPopup.FindControl("ResultsPanel")
		Dim noResultsPanel = SearchPopup.FindControl("NoResultsPanel")
		If results.Count > 0 Then
			resultsPanel.Visible = True
			noResultsPanel.Visible = False
			Dim grouppedResults = results.GroupBy(Function(sr) sr.Product)
			For Each productGroup In grouppedResults
				Dim product = productGroup.Key
				Dim group = SearchResultsNavBar.Groups.Add(product.NavItemTitle.ToUpper(), String.Empty, String.Empty, Utils.GenerateDemoUrl(product.Intro))
				For Each res In productGroup
					group.Items.Add(res.Text, String.Empty, String.Empty, Utils.GenerateDemoUrl(res.Demo))
				Next res
			Next productGroup
		Else
			resultsPanel.Visible = False
			noResultsPanel.Visible = True
			Dim requestText = TryCast(noResultsPanel.FindControl("RequestText"), HtmlGenericControl)
			requestText.InnerHtml = text
		End If
	End Sub
End Class
